hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/ReplicationManager.java

```diff
@@ -448,6 +436,153 @@ protected ContainerHealthResult processContainer(ContainerInfo containerInfo,
     return health;
   }
 
+  private ContainerHealthResult checkContainerState(
+      final ContainerInfo containerInfo, final ReplicationManagerReport report)
+      throws ContainerNotFoundException {
+    ContainerID containerID = containerInfo.containerID();
+    Set<ContainerReplica> replicas = containerManager
+        .getContainerReplicas(containerID);
+    HddsProtos.LifeCycleState state = containerInfo.getState();
+    ContainerHealthResult health =
+        new ContainerHealthResult.HealthyResult(containerInfo);
+
+    if (state == HddsProtos.LifeCycleState.OPEN) {
```

Comment:
Nit: Since each state has a separate handler, maybe `switch` would be a better fit.